src/Std/Data/Internal/List/Associative.lean

```diff
@@ -5092,6 +5407,285 @@ theorem getEntry?_map [BEq α] [EquivBEq α]
+    getKey k (List.filter (fun p => containsKey p.fst l₂) l₁) h₁ = getKey k l₁ h₂ := by
+  suffices some (getKey k (List.filter (fun p => containsKey p.fst l₂) l₁) h₁) = some (getKey k l₁ h₂) by
+    injections
+  simp only [← getKey?_eq_some_getKey]
+  apply List.getKey?_filter_containsKey_of_containsKey_right dl₁
+  · rw [List.containsKey_filter_containsKey_iff] at h₁
+    · exact h₁.2
+    · exact dl₁
+
+theorem List.containsKey_filter_containsKey_eq_false_of_containsKey_eq_false_left [BEq α] [EquivBEq α] {l₁ l₂ : List ((a : α) × β a)} {hl₁ : DistinctKeys l₁} {k : α} :
+    containsKey k l₁ = false → containsKey k (List.filter (fun p => containsKey p.fst l₂) l₁) = false := by
+  intro h
+  rw [containsKey_filter_containsKey]
+  . simp [h]
+  . exact hl₁
+
+theorem List.containsKey_filter_containsKey_eq_false_of_containsKey_eq_false_right [BEq α] [EquivBEq α] {l₁ l₂ : List ((a : α) × β a)} {hl₁ : DistinctKeys l₁} {k : α} :
+    containsKey k l₂ = false → containsKey k (List.filter (fun p => containsKey p.fst l₂) l₁) = false := by
+  intro h
+  rw [containsKey_filter_containsKey]
+  . simp [h]
+  . exact hl₁
+
+theorem getValue?_filter_containsKey_eq_if_containsKey {β : Type v} [BEq α] [EquivBEq α]
+    {l₁ l₂ : List ((_ : α) × β)} {k : α}
+    (dl₁ : DistinctKeys l₁) :
+    getValue? k (List.filter (fun p => containsKey p.fst l₂) l₁) =
+    if containsKey k l₂ = true then getValue? k l₁ else none := by
+  simp only [getValue?_eq_getEntry?]
+  split
+  case isTrue h =>
+    congr 1
+    rw [getEntry?_filter]
+    . simp only [Option.filter]
+      split
+      case h_1 _ val heq =>
+        simp [heq]
+        have := PartialEquivBEq.symm <| @beq_of_getEntry?_eq_some α (fun _ => β)  _ l₁ k val heq
+        rw [← containsKey_congr this]
+        exact h
+      case h_2 _ heq =>
+        simp [heq]
+    . exact dl₁
+  case isFalse h =>
+    simp only [Option.map_eq_none_iff, getEntry?_eq_none]
+    simp only [Bool.not_eq_true] at h
+    apply List.containsKey_filter_containsKey_eq_false_of_containsKey_eq_false_right
+    . exact dl₁
+    . exact h
+
+theorem getValue?_filter_containsKey_eq_containsKey {β : Type v} [BEq α] [EquivBEq α]
+    {l₁ l₂ : List ((_ : α) × β)} {k : α}
+    (dl₁ : DistinctKeys l₁) :
+    containsKey k l₂ → getValue? k (List.filter (fun p => containsKey p.fst l₂) l₁) =
+      getValue? k l₁ := by
+  intro h
+  rw [getValue?_filter_containsKey_eq_if_containsKey, h]
+  · simp only [↓reduceIte]
+  · exact dl₁
+
+theorem getValue?_filter_of_not_contains {β : Type v} [BEq α] [EquivBEq α]
+    {l₁ l₂ : List ((_ : α) × β)} {k : α}
+    (dl₁ : DistinctKeys l₁) :
+    containsKey k l₁ = false → getValue? k (List.filter (fun p => containsKey p.fst l₂) l₁) = none := by
+  intro h
+  rw [getValue?_filter_containsKey_eq_if_containsKey]
+  · split
+    · rw [@getValue?_eq_none α β _ l₁ k]
+      exact h
+    · rfl
+  · exact dl₁
+
+theorem List.getValue?_filter_of_containsKey_eq_false_right {β : Type v} [BEq α] [EquivBEq α]
```

Comment:
You are adding a lot of theorems with names like `Std.Internal.List.List.getValue?_filter_of_containsKey_eq_false_right` (note the double `List` namespace) here.